packages/flutter/test/widgets/page_view_test.dart

```diff
@@ -346,26 +372,49 @@ void main() {
 
     Widget build(PageController controller) {
       return new PageView.builder(
-        controller: controller,
-        itemCount: kStates.length,
-        itemBuilder: (BuildContext context, int index) {
-          return new Container(
-            height: 200.0,
-            color: index % 2 == 0 ? const Color(0xFF0000FF) : const Color(0xFF00FF00),
-            child: new Text(kStates[index]),
-          );
-        }
+              controller: controller,
+              itemCount: kStates.length,
+              itemBuilder: (BuildContext context, int index) {
+                return new Container(
+                  height: 200.0,
+                  color: index % 2 == 0 ? const Color(0xFF0000FF) : const Color(
+                          0xFF00FF00),
+                  child: new Text(kStates[index]),
+                );
+              }
       );
     }
 
     await tester.pumpWidget(build(controller));
 
     expect(tester.getTopLeft(find.text('Alabama')), const Point(-100.0, 0.0));
-    expect(tester.getBottomRight(find.text('Alabama')), const Point(900.0, 600.0));
+    expect(tester.getBottomRight(find.text('Alabama')),
+            const Point(900.0, 600.0));
 
     controller.jumpToPage(10);
     await tester.pump();
 
     expect(tester.getTopLeft(find.text('Hawaii')), const Point(-100.0, 0.0));
   });
-}
+
+  testWidgets('PageView does not report page changed on overscroll', (
+    WidgetTester tester) async {
+      PageController controller = new PageController(initialPage: kStates.length - 1);
+      int changeIndex = 0;
+      Widget build() {
+        return new PageView(
+          children: kStates.map<Widget>((String state) => new Text(state))
+                  .toList(),
+          controller: controller,
+          onPageChanged: (page) {
+            changeIndex = page;
+          }
+        );
+      }
+
+      await tester.pumpWidget(build());
+      controller.jumpToPage(kStates.length * 2);  // try to move beyond max range
+      expect(changeIndex, 0);   // change index should be zero, shouldn't fire onPageChanged
```

Comment:
I'd put in an `await tester.pump()` and another `expect(changeIndex, 0);` just to make sure the notification doesn't wait for the pump.  (It doesn't, but that makes a more complete test.)